src/ptvsd/adapter/debuggee.py

```diff
@@ -58,11 +58,13 @@ def spawn_and_connect(request):
     Caller is responsible for calling start() on the returned channel.
     """
 
-    channel = channels.Channels().accept_connection_from_server(
-        ("127.0.0.1", 0),
-        before_accept=lambda address: _parse_request_and_spawn(request, address),
-    )
-    return channel
+    if request("noDebug", json.default(False)):
+        _parse_request_and_spawn(request, (None, None))
```

Comment:
It might be better to pass `None` here to indicate that it's not an address at all. And then in `_parse_request`, it should deconstruct `address` into `host, port` only in the branch where it needs it for the command line args.